### PR TITLE
Default values for stuct members and scalar initialization of array members

### PR DESCRIPTION
**Pull request by rohandkn**
| | |
|:---|:---|
|**_Originating Project/Creator_**| F Prime |
|**_Affected Component_**| Autocoder |
|**_Affected Architectures(s)_**| serializable |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**| Y  |
|**_Builds Without Errors (y/n)_**| Y  |
|**_Unit Tests Pass (y/n)_**| Y |
|**_Documentation Included (y/n)_**| Y |

---
## Change Description

Added support to allow a nested default element for struct members in F' XML. Autocoder initializes struct member to this  
value in the constructor for the serializable. Also, allowed for initialization of arrays with a scalar value. This sets all elements in the array to the scalar value.
 
## Rationale

These changes to the autocoder were made to support the FPP syntax and features.

## Testing/Review Recommendations

Created new serialize4 test which tests autocoder functionality as it pertains to serializable objects.

All tests built and run with no issues.

## Future Work

N/A

### CODE REVIEW

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
@bocchino want to review this too?

File: docs/UsersGuide/dev/xml-specification.md [hunk 1]
```diff
@@ -609,7 +612,7 @@ specification.
...
-| commands                   |                 | Optional. Starts the section where software commands are defined.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |                                                                                                                                                                        |
```
Comment (bocchino) [inline docs/UsersGuide/dev/xml-specification.md:612]: @rohandkn I think this deletion is not intended. Can you revert it?

**Review by bocchino — CHANGES_REQUESTED** [review]
One small change request, otherwise looks good!

File: docs/UsersGuide/dev/xml-specification.md (hunk 1 above)
Comment (rohandkn) [reply docs/UsersGuide/dev/xml-specification.md]: Yes, made the change.

**Review by bocchino — APPROVED** [review]
Looks good!

**Pull request merged**